zenoh/src/net/routing/interceptor/authorization.rs

```diff
@@ -177,6 +177,18 @@ impl PolicyEnforcer {
                                 );
                             }
                         }
+                        match rule.usernames {
```

Comment:
What's the point of having `rule.usernames` as `Option<Vec<_>>` if `None` is set to `Some` with an empty vector? Couldn't we just use a `Vec<_>`?

In any case, the `match` could be rewritten as:
```rust
if rule.usernames.is_none() {
    rule.usernames = Some(Vec::new());
}
```

Reply:
`Option<Vec<_>>` is just to allow the user to not specify it in the config, instead of having to explicitely set it to an empty vector when the filter is not a concern for the rule in question.